Sources/NIO/Codec.swift

```diff
@@ -101,7 +126,31 @@ private extension ChannelHandlerContext {
     }
 }
 
-extension ByteToMessageDecoder {
+public class ByteToMessageHandler<Decoder: ByteToMessageDecoder>: ChannelInboundHandler {
+    public typealias InboundIn = ByteBuffer
+    public typealias InboundOut = Decoder.InboundOut
+
+    internal private(set) var decoder: Decoder
```

Comment:
Is there any reason for the decoder to be `internal`?

Reply:
tests... Also to be thought through